src/fe-small-advertisements/src/actions/name.ts

```diff
@@ -0,0 +1,16 @@
+export interface User {
+  name: string;
+}
+
+export const CHANGE_NAME = 'CHANGE_NAME';
+
+export interface ChangeNameAction {
+  type: typeof CHANGE_NAME;
+  payload: User;
+}
+
+export const changeNameActionCreator = ({
+  name,
+}: {
+  name: string;
+}): ChangeNameAction => ({ type: CHANGE_NAME, payload: { name } });
```

Comment:
```suggestion
}: User): ChangeNameAction => ({ type: CHANGE_NAME, payload: { name } });
```
isnt this `User`?

Reply:
Yes, yes! I am telling you, Typescript in React still looks like a completely different language to me. I refactored the whole thing now.